cmd/generate.go

```diff
@@ -28,155 +28,165 @@ var generateCmd = &cobra.Command{
 				if verbose {
-					fmt.Printf("No token found. Attempting to generate config without one...\n")
+					fmt.Printf("No token found. Attempting to generate conf without one...\n")
 				}
 			}
 		}
 
 		// use cert path from cobra if empty
-		if config.CertPath == "" {
-			config.CertPath = cacertPath
+		if conf.CertPath == "" {
+			conf.CertPath = cacertPath
 		}
 
-		// show config as JSON and generators if verbose
+		// show conf as JSON and generators if verbose
 		if verbose {
-			b, err := json.MarshalIndent(config, "", "  ")
+			b, err := json.MarshalIndent(conf, "", "  ")
 			if err != nil {
-				fmt.Printf("failed to marshal config: %v\n", err)
+				fmt.Printf("failed to marshal conf: %v\n", err)
 			}
 			fmt.Printf("%v\n", string(b))
 		}
 
 		// run all of the target recursively until completion if provided
 		if len(targets) > 0 {
-			RunTargets(&config, args, targets...)
+			RunTargets(&conf, args, targets...)
 		} else {
 			if pluginPath == "" {
 				fmt.Printf("no plugin path specified")
 				return
 			}
 
+			// load the templates to use
+			templates := map[string]generator.Template{}
+			for _, path := range templatePaths {
+				template := generator.Template{}
+				template.LoadFromFile(path)
+				if !template.IsEmpty() {
+					templates[path] = template
+				}
+			}
+
+			params := generator.Params{
+				Templates: templates,
+			}
+
 			// run generator.Generate() with just plugin path and templates provided
-			generator.Generate(&config, generator.Params{
-				PluginPath:    pluginPath,
-				TemplatePaths: templatePaths,
-			})
+			outputBytes, err := generator.Generate(pluginPath, params)
+			if err != nil {
+				log.Error().Err(err).Msg("failed to generate files")
+			}
 
+			// if we have more than one target and output is set, create configs in directory
+			writeOutput(outputBytes, len(targets), len(outputMap))
 		}
 	},
 }
 
 // Generate files by supplying a list of targets as string values. Currently,
-// targets are defined statically in a config file. Targets are ran recursively
+// targets are defined statically in a conf file. Targets are ran recursively
 // if more targets are nested in a defined target, but will not run additional
 // child targets if it is the same as the parent.
 //
 // NOTE: This may be changed in the future how this is done.
-func RunTargets(config *configurator.Config, args []string, targets ...string) {
-	// generate config with each supplied target
+func RunTargets(conf *config.Config, args []string, targets ...string) {
+	// generate conf with each supplied target
 	for _, target := range targets {
-		outputBytes, err := generator.GenerateWithTarget(config, generator.Params{
-			Args:       args,
-			PluginPath: pluginPath,
-			Target:     target,
-			Verbose:    verbose,
-		})
+		outputBytes, err := generator.GenerateWithTarget(conf, target)
 		if err != nil {
-			log.Error().Err(err).Msg("failed to generate config")
+			log.Error().Err(err).Msg("failed to generate conf")
```

Comment:
As far as the comments and print statements are concerned, I think it would be clearer to the user to keep "config" instead of changing to "conf".

Reply:
Same as [this comment](https://github.com/OpenCHAMI/configurator/pull/24#discussion_r1868297982).